File: src/Omnia.ProcessManagement.Models/ShapeTemplates/ShapeTemplate.cs
using Omnia.Fx.Models.Language;
using System;
using System.Collections.Generic;
using System.Text;

namespace Omnia.ProcessManagement.Models.ShapeTemplates
{
    public class ShapeTemplate
    {
        public ShapeTemplate()
        {
            Id = Guid.NewGuid();
        }
        public Guid Id { get; set; }
        public MultilingualString Title { get; set; }
        public bool BuiltIn { get; set; }
        public ShapeTemplateSettings Settings { get; set; }
    }
}

File: src/Omnia.ProcessManagement.Web/client/fx/models/data/shapetemplates/ShapeTemplate.ts
import { GuidValue, Guid, MultilingualString } from '@omnia/fx-models';
import { ShapeTemplateSettings } from './ShapeTemplateSettings';
import { ShapeTemplateType } from '../enums/Enums';

export interface ShapeTemplate {
    id: GuidValue,
    title: MultilingualString,
    settings: ShapeTemplateSettings,
    builtIn: boolean,

    //client-side
    multilingualTitle: string
}

/**
 * NOTE: whenever an new property is added in settings, we need to define its initial value in factory
 * To ensure it fully react on view
 * */
export const ShapeTemplateFactory = {
    createDefaultShapeTemplate(): ShapeTemplate {
        let shapeTemplate: ShapeTemplate = {
            id: Guid.newGuid(),
            title: { isMultilingualString: true },
            settings: {
                type: ShapeTemplateType.FreeformShape
            },
            builtIn: false
        } as ShapeTemplate

        return shapeTemplate;
    }
}